app/jobs/sync_publisher_statement_job.rb

```diff
@@ -1,14 +1,32 @@
 class SyncPublisherStatementJob < ApplicationJob
   queue_as :default
 
-  def perform(publisher_statement_id:)
+  def perform(publisher_statement_id:, first_attempt: nil)
+    if first_attempt
+      time_elapsed = Time.now.to_i - first_attempt
+      return if time_elapsed > 3.minutes
+    else
+      first_attempt = Time.now.to_i
+      time_elapsed = 0
+    end
+
     publisher_statement = PublisherStatement.find(publisher_statement_id)
 
     PublisherStatementSyncer.new(publisher_statement: publisher_statement).perform
 
     publisher_statement.reload
+
     unless publisher_statement.contents.present?
-      SyncPublisherStatementJob.set(wait: 3.seconds).perform_later(publisher_statement_id: publisher_statement.id)
+      if time_elapsed < 20.seconds
```

Comment:
minor: this might be cleaner if extracted in a small private method

Reply:
ok, will do